### DISCUSSION TITLE
Unexpected behavior QSelect+QExpansionItem

### DISCUSSION THREAD

**Discussion by Drabuna** [quasarframework/quasar#13721] [General - Components / Directives / etc] [answered]
### What happened?

When opening a QSelect with a bunch of QExpansionItems and some child elements, `q-virtual-scroll__padding`  starts behaving weirdly. 
I spent quite a lot of time searching for why, and the only reasonable advice was to add `q-virtual-scroll--with-prev` class to child elements, which didn't help at all. 
The thing that did help was to set `virtual-scroll-slice-size` to the total number of elements (36 for the provided example).
The documentation for that is not obvious at all, and only randomly I've discovered how to resolve this.
Is this the expected behavior for the virtual scroll?

![image](https://user-images.githubusercontent.com/54286/174191878-7712bf69-2703-44b4-8611-61ea0d95f43e.png)

### What did you expect to happen?

Expected normal-sized QSelect menu, without `q-virtual-scroll__padding` randomly changing size.

### Reproduction URL

https://codepen.io/drabuna/pen/Jjpzvrx

### How to reproduce?

1. Go to the provided link
2. Click on the q-select
3. Be amazed by unnecessary large `q-virtual-scroll__padding`

### Flavour

Quasar CLI with Webpack (@quasar/cli | @quasar/app-webpack)

### Areas

Components (quasar)

### Platforms/Browsers

Chrome, Safari

**Comment by pdanpdan** [comment, marked as answer]
In your case you have a large list of options that do not get rendered, but the scroll expects the items.

You can work around it like https://codepen.io/pdanpdan/pen/abqMRLz?editors=1010 but it's a bad approach.

Better go with something like this: https://codepen.io/pdanpdan/pen/ZErPqjg?editors=1010 (adapted from Jeff Galbraith)

You can find more resources https://www.google.com/search?q=jeff+quasar+qselect+expansion
